### PR TITLE
fix share algo path issue

### PR DESCRIPTION
**Pull request by Andyli1007**

### CODE REVIEW

File: src/misc/msccl/msccl_lifecycle.cc [hunk 1]
```diff
@@ -103,21 +107,31 @@ static ncclResult_t mscclInternalSchedulerInit() {
     mscclAlgoDir = mscclAlgoDirStr.c_str();
     // Get share Directory Paths
     mscclAlgoShareDirStr = selfLibPath.substr(0, selfLibPath.find_last_of("/\\") + 1);
-    mscclAlgoShareDirStr += (mscclUnitTestMode && mscclUnitTestMode()) ? mscclUnitTestAlgoShareDirPath : mscclAlgoShareDirPath;
+    mscclAlgoShareDirStr += (mscclUnitTestMode && mscclUnitTestMode()) ? mscclUnitTestAlgoShareDirPath : mscclAlgoShareDirPath;
```
Comment (yzygitzh) [inline src/misc/msccl/msccl_lifecycle.cc:110]: Remove ending space

File: src/misc/msccl/msccl_lifecycle.cc [hunk 2]
```diff
@@ -103,21 +107,31 @@ static ncclResult_t mscclInternalSchedulerInit() {
     mscclAlgoDir = mscclAlgoDirStr.c_str();
     // Get share Directory Paths
     mscclAlgoShareDirStr = selfLibPath.substr(0, selfLibPath.find_last_of("/\\") + 1);
-    mscclAlgoShareDirStr += (mscclUnitTestMode && mscclUnitTestMode()) ? mscclUnitTestAlgoShareDirPath : mscclAlgoShareDirPath;
+    mscclAlgoShareDirStr += (mscclUnitTestMode && mscclUnitTestMode()) ? mscclUnitTestAlgoShareDirPath : mscclAlgoShareDirPath; 
     mscclAlgoShareDir = mscclAlgoShareDirStr.c_str();
+
+    mscclPackageInstalledAlgoShareDirStr = (mscclUnitTestMode && mscclUnitTestMode()) ? mscclUnitTestPackageInstalledAlgoShareDirPath : mscclPackageInstalledAlgoShareDirPath;
```
Comment (yzygitzh) [inline src/misc/msccl/msccl_lifecycle.cc:113]: Remove ending space

**Review by yzygitzh — APPROVED** [review]

File: src/misc/msccl/msccl_lifecycle.cc (hunk 2 above)
Comment (Andyli1007) [reply src/misc/msccl/msccl_lifecycle.cc]: removed.

File: src/misc/msccl/msccl_lifecycle.cc (hunk 1 above)
Comment (Andyli1007) [reply src/misc/msccl/msccl_lifecycle.cc]: removed.

**Pull request merged**